src/features/form-page/ui/Form.tsx

```diff
@@ -0,0 +1,154 @@
+import classNames from 'classnames/bind';
+import * as Yup from 'yup';
+
+import styles from './Form.module.scss';
+import { ChangeEvent, FormEvent, useState } from 'react';
+import Input from '@/src/shared/@common/ui/input';
+
+const cx = classNames.bind(styles);
+
+const formSchema = Yup.object({
+  email: Yup.string()
+    .trim()
+    .email('이메일 형식으로 입력해 주세요.')
+    .required('이메일을 입력해 주세요.'),
+  password: Yup.string()
+    .trim()
+    .min(8, '8자 이상 입력해 주세요.')
+    .required('비밀번호를 입력해 주세요.'),
+  passwordConfirm: Yup.string()
+    .trim()
+    .oneOf([Yup.ref('password')], '비밀번호가 일치하지 않습니다.')
+    .required('비밀번호를 입력해 주세요.'),
+});
+
+interface FormData {
+  [key: string]: {
+    value: string;
+    isError: boolean;
+    message: string;
+  };
+}
+
+const initialFormData: FormData = {
+  email: {
+    value: '',
+    isError: false,
+    message: '',
+  },
+  password: {
+    value: '',
+    isError: false,
+    message: '',
+  },
+  passwordConfirm: {
+    value: '',
+    isError: false,
+    message: '',
+  },
+};
+
+export default function Form() {
+  const [formData, setFormData] = useState<FormData>(initialFormData);
+
+  const handleSubmit = async (e: FormEvent) => {
+    e.preventDefault();
+
+    const data = {
+      email: formData.email.value,
+      password: formData.password.value,
+      passwordConfirm: formData.passwordConfirm.value,
+    };
+
+    for (const key in formData) {
+      try {
+        await formSchema.validateAt(key, data);
+
+        setFormData((prev) => ({
+          ...prev,
+          [key]: { ...prev[key], isError: false, message: '' },
+        }));
+      } catch (error) {
+        if (error instanceof Yup.ValidationError) {
+          setFormData((prev) => ({
+            ...prev,
+            [key]: { ...prev[key], isError: true, message: error.message },
+          }));
+          return;
+        }
+      }
+    }
```

Comment:
해당 코드를 아래의 코드를 이용해서 개선해 보심이 어떨까요?!
```suggestion
formSchema.validate(data, { abortEarly: false }).catch((error) => {
  return Object.fromEntries(
    error.inner.map((err: Yup.ValidationError) => [err.path, err.message]),
  );
});
```